src/content/docs/en/guides/upgrade-to/v6.mdx

```diff
@@ -585,6 +586,124 @@ If you need more control over environment variables in Astro, we recommend you u
 
 <ReadMore>Learn more about [environment variables](/en/guides/environment-variables/) in Astro, including `astro:env`.</ReadMore>
 
+### Changed: Markdown heading ID generation
+
+<SourcePR number="14494" title="feat!: stabilize experimental.headingIdCompat"/>
+
+In Astro 5.5, the `experimental.headingIdCompat` flag was introduced to make the IDs generated by Astro for Markdown headings compatible with common platforms like GitHub and npm, using the popular [`github-slugger`](https://github.com/Flet/github-slugger) package.
+
+In Astro 5.x, Astro had an additional default processing step that stripped a trailing hyphen from the end of IDs for headings ending in special characters.
+
+Astro 6.0 removes this experimental flag and makes this the new default behavior in Astro: trailing hyphen from the end of IDs for headings ending in special characters isn't stripped anymore.
```

Comment:
There's a weird mix of singular and plural here. I guess we can go with plural all the way and make it flow better?
```suggestion
In Astro 5.x, Astro had an additional default processing step that stripped trailing hyphens from the end of IDs for headings ending in special characters.

Astro 6.0 removes this experimental flag and makes this the new default behavior in Astro: trailing hyphens from the end of IDs for headings ending in special characters aren't stripped anymore.
```